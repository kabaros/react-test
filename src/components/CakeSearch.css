.cake-search input {
    border: 1px solid #c7c7c7;
    margin: 10px;
    width: 80%;
    padding: 10px;
    font-size: 1.2em;
}