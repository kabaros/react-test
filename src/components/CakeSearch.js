import React from 'react';
import PropTypes from 'prop-types';
import './CakeSearch.css';

const CakeSearch = ({ onSearch }) => {
    return(<div className="cake-search">
        <input
         type="text"
         placeholder="Search for a cake"
         onChange={({target}) => onSearch(target.value)} />
    </div>)
};

export default CakeSearch;