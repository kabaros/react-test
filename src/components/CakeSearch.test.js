import React from 'react';
import renderer from 'react-test-renderer';
import CakeSearch from './CakeSearch';

it('renders a Search box for matching cakes', () => {
  const cakeFinder = () => true;
  const component = renderer.create(
    <CakeSearch onSearch={cakeFinder} />
  );
  expect(component).toMatchSnapshot();
});
