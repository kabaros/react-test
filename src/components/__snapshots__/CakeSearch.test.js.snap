// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Search box for matching cakes 1`] = `
<div
  className="cake-search"
>
  <input
    onChange={[Function]}
    placeholder="Search for a cake"
    type="text"
  />
</div>
`;
